backend/chainlit/server.py

```diff
@@ -449,7 +449,7 @@ def _get_oauth_redirect_error(error: str) -> Response:
     )
     response = RedirectResponse(
         # FIXME: redirect to the right frontend base url to improve the dev environment
-        url=f"/login?{params}",  # Shouldn't there be {root_path} here?
+        url=f"{config.run.root_path.rstrip('/')}/login?{params}",
```

Comment:
Instead of writing the URL, including `root_path` manually, I would suggest using [`request.url_for()`](https://fastapi.tiangolo.com/reference/request/?h=url_for#fastapi.Request.receive) to generate the URL with root path automatically.

Yes, that does mean we add a request parameter there, but it also means less custom code and so less maintenance and bugs in the future!